contracts/stellar-example/src/tests/test.rs

```diff
@@ -180,78 +180,199 @@ fn gmp_example() {
+
+    let destination_chain = String::from_str(&env, DESTINATION_CHAIN_NAME);
+    let TestConfig {
+        signers: destination_signers,
+        gateway_client: destination_gateway_client,
+        its_client: destination_its_client,
+        app: destination_app,
         ..
-    } = setup_app(&env);
-    let source_chain = its_client.its_hub_chain_name();
-    let source_address: String = its_client.its_hub_address();
+    } = setup_app(&env, Some(destination_chain.clone()));
 
-    let amount = 1000;
-    let token_id = setup_its_token(&env, &its_client, &Address::generate(&env), amount);
+    let transfer_amount = 1000;
 
-    let original_source_chain = String::from_str(&env, "ethereum");
-    its_client
+    let (token_id, token_metadata) =
+        setup_its_token(&env, &original_source_its_client, &user, transfer_amount);
+    let token_address = original_source_its_client.token_address(&token_id);
+
+    let gas_token_contract = env.register_stellar_asset_contract_v2(user.clone());
+    let gas_amount = 50;
+    let gas_token = Token {
+        address: gas_token_contract.address(),
+        amount: gas_amount,
+    };
+    StellarAssetClient::new(&env, &gas_token.address)
+        .mock_all_auths()
+        .mint(&user, &gas_token.amount);
+
+    original_source_its_client
+        .mock_all_auths()
+        .set_trusted_chain(&destination_chain);
+
+    let original_source_trusted_chain_set_event =
+        events::fmt_last_emitted_event::<TrustedChainSetEvent>(&env);
+
+    destination_its_client
         .mock_all_auths()
         .set_trusted_chain(&original_source_chain);
 
-    let trusted_chain_set_event = events::fmt_last_emitted_event::<TrustedChainSetEvent>(&env);
+    let destination_trusted_chain_set_event =
+        events::fmt_last_emitted_event::<TrustedChainSetEvent>(&env);
 
     let data = Address::generate(&env).to_string_bytes();
 
-    let msg = stellar_interchain_token_service::types::HubMessage::ReceiveFromHub {
+    original_source_its_client
+        .mock_all_auths()
+        .register_canonical_token(&token_address);
+
+    original_source_its_client
+        .mock_all_auths()
+        .deploy_remote_canonical_token(&token_address, &destination_chain, &user, &gas_token);
+
+    let token_asset_client = StellarAssetClient::new(&env, &gas_token.address);
+    token_asset_client
+        .mock_all_auths()
+        .mint(&user, &transfer_amount);
+
+    // DeployInterchainToken execute
+    let deploy_msg = stellar_interchain_token_service::types::HubMessage::ReceiveFromHub {
+        source_chain: original_source_chain.clone(),
+        message: stellar_interchain_token_service::types::Message::DeployInterchainToken(
+            stellar_interchain_token_service::types::DeployInterchainToken {
+                token_id: token_id.clone(),
+                name: token_metadata.name.clone(),
+                symbol: token_metadata.symbol.clone(),
+                decimals: token_metadata.decimal as u8,
```

Comment:
query this via the token::Interface, instead of needing the setup_its_token helper to return token metadata everywhere